packages/contracts-router/contracts/test/Router.t.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.7.6;
+
+import "forge-std/Test.sol";
+import {RouterHarness} from "./harnesses/RouterHarness.sol";
+
+contract RouterTest is Test {
+    RouterHarness router;
+
+    uint32 testDomain;
+    bytes32 testRouter;
+
+    function setUp() public {
+        testDomain = 123213;
+        testRouter = "router address";
+        router = new RouterHarness();
+        address xAppConnectionManager = vm.addr(1321);
+        router.exposed__XAppConnectionClient_initialize(xAppConnectionManager);
+        router.enrollRemoteRouter(testDomain, testRouter);
+    }
+
+    function test_handle() public {
```

Comment:
i don't think we need to test virtual functions by implementing a fake version and testing that 🤔 idk. it's like testing an interface by making random implementations for the functions - i don't think it's really helpful